aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2650,18 +2660,48 @@ Tensor& linalg_matrix_norm_out(
 
 // Numerical or None norms
 Tensor linalg_norm(const Tensor& self, const optional<Scalar>& opt_ord, optional<IntArrayRef> opt_dim, bool keepdim, optional<ScalarType> opt_dtype) {
+  int64_t ndim = self.dim();
+  std::vector<int64_t> dim_ = opt_dim.has_value() ? opt_dim.value().vec() : make_dim_list(ndim);
+  if (!opt_ord.has_value() || dim_.size() == 1) {
+    return at::linalg_vector_norm(self, opt_ord.value_or(2), opt_dim, keepdim, opt_dtype);
+  }
+
   auto options = TensorOptions().dtype(opt_dtype.has_value() ? opt_dtype.value() : toRealValueType(self.scalar_type())).device(self.device());
   Tensor result = at::empty({0}, options);
-  return at::native::linalg_norm_out(
-      self, opt_ord, opt_dim, keepdim, opt_dtype, result);
+
+ if (dim_.size() == 2) {
+    _linalg_norm_matrix_out(result, self, opt_ord.value(), dim_, keepdim, opt_dtype);
```

Comment:
Why does it call the out variant?

Reply:
This function is what implements the matrix norm. It is an internal function (not ATEN) and is just a 'function call' that does not get registered into autograd.